schema/crdb/tuf-artifact-replication/up02.sql

```diff
@@ -0,0 +1,4 @@
+ALTER TABLE omicron.public.sled
+    ADD COLUMN IF NOT EXISTS repo_depot_port INT4
+        CHECK (port BETWEEN 0 AND 65535)
+        NOT NULL DEFAULT 0;
```

Comment:
We are relying on `sled_upsert`, yeah. I am not actually certain if that runs every time Sled Agent starts, I did not look that closely — it seems that there's a background task that periodically notifies Nexus of Sled Agent's existence?

Reply:
So if anyone tries using this column before that value propagates, they'll try communicating with port zero on the sled agent, right?

Should we set this value to `12348` instead of zero? It's a constant anyway, that seems like it would avoid this period of "seeing repot_depo_port = 0"

Reply:
12348 is the port in use, it's a const:

https://github.com/oxidecomputer/omicron/blob/dcc0df384de6b1c708f715f66cc013c6fc3cde5c/common/src/address.rs#L31-L32

As noted elsewhere the main reason this is in the database is to support tests, where the const is not used. So I think a 12348 default for current sleds is relatively safe. (Making this nullable instead I think would be the right thing if we were rolling into this upgrade instead of doing mupdate.)

Reply:
To be clear: the assumption here is that any system on which the migration is running is a "real" system and for those the port has always been 12348 and so filling it in this way is definitely correct.  That makes sense to me.  Add a comment to that effect?